torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -126,7 +161,29 @@ void PropagateShapeOnNode(Node * node) {
       node->output()->setType(tp->withSizesStrides(sizes, strides));
     } break;
     case kview: {
-      node->output()->setType(types.at(0)->withSizes(node->is(ksizes)));
+      JIT_ASSERT(types.size() == 1);
+      auto sizes = node->is(ksize);
+      bool inferred = false;
+      size_t inferred_idx;
+      int64_t size_product = 1;
+      for (size_t i=0; i<sizes.size(); ++i) {
+        if (sizes[i] == -1) {
+          if (inferred)
+            throw std::runtime_error("-1 occured in more than 1 size position");
```

Comment:
We should really let those things fail at run time, and generate DYNAMIC type here. Otherwise we'll be in the business of maintaining the same error messages twice

Reply:
Agreed. But for now this is consistent with the rest of the implementation in this file.